templates/init.erb

```diff
@@ -27,77 +69,73 @@ start() {
   ulimit -n 65536
   ulimit -s 10240
   ulimit -c unlimited
-  if [ -f $PID_FILE ]
-  then
-    PID=`cat $PID_FILE`
-    if [ ! -z "`ps -ef | awk '{print $2}' | grep "^$PID$"`" ]
-    then
+  if [ -f "$PID_FILE" ]; then
+    PID=`cat "$PID_FILE"`
+    if [ `ps -p "$PID" -o pid= || echo 1` -eq `pgrep -f "$PGREP_PATTERN" || echo 2` ] ; then
```

Comment:
yes!! it works really well

what it does is this:
if `ps -p 2194 -o pid=` doesn't find that pid to be running, it'll **output** nothing, but **return** 1. In that case the `|| echo 1` case will fire. simiarily, if `pgrep -f kafka.Kafka` doesn't find anything, it will **output** nothing, but **return** 1. Now, `|| echo 2` is the result. hence, we're comparing
`[ 1 -eq 2 ]`, which will be false. but! if one of `ps` or `pgrep` returns a sensible value, something is off! and we're catching those cases, too: `[ 2967 -eq 2 ]` is just as false, as `[ 1 -eq 2957 ]` is.